website/blog/2023-02-21.ferretdb-compare-arrays.md

```diff
@@ -0,0 +1,530 @@
+---
+title: "How MongoDB compares arrays in filtering"
+slug: ferretdb-compare-arrays
+authors: [chi]
+description: In this article, we explore in detail how MongoDB BSON arrays are compared
+in filter operators.
+keywords:
+  - MongoDB sorting
+  - MongoDB array comparison
+image: /img/blog/ferretdb-how-mongodb-compares-arrays.jpg
+unlisted: true
+---
+
+In this article, we explore in detail how MongoDB BSON arrays are compared in filter operators.
+
+![How MongoDB compares arrays](/img/blog/ferretdb-how-mongodb-compares-arrays.jpg)
+
+<!--truncate-->
+
+## Understanding BSON Comparison for Filtering
+
+The purpose of comparing BSON values is to determine which value is
+equal, greater or less than the other.
+The comparison is used in various operations, such as sorting, filter operators,
+as well as update operators.
+When comparing values of different BSON types, the [BSON Comparison Order](#bson-comparison-order)
+is used to determine the result.
+However, in the case of array comparison against other BSON value,
+the comparison may use the BSON comparison order or an element from the array.
```

Comment:
I don't really understand this sentence - "the comparison may use the BSON comparison order or an element from the array" -- I guess it means it either compare array according to its place in the BSON comparison order or according to elements of the array... I'm not sure how to phrase it better though.

I see below there is this sentence:

> The BSON comparison order is used for comparing different BSON types. A BSON array may use an element from an array to compare with other BSON type. 

and this second sentence looks clear to me. So maybe we can only use the second sentence if they mean the same?

Reply:
Thanks for pointing this out. Let me handle this.